src/components/Dashboard/TodoList.vue

```diff
@@ -1403,18 +1403,33 @@ export default class TodoList extends Vue {
         // see also ItemHeaderActions.vue:correctThisFiling()
         switch (item.correctedFilingType) {
           case FilingNames.ALTERATION:
-          case FilingNames.INCORPORATION_APPLICATION:
           case FilingNames.CHANGE_OF_REGISTRATION:
           case FilingNames.CORRECTION:
+          case FilingNames.INCORPORATION_APPLICATION:
           case FilingNames.REGISTRATION:
             // resume correction via Edit UI
             const correctionUrl = `${this.getEditUrl}${this.getIdentifier}/correction/?correction-id=${item.filingId}`
             navigate(correctionUrl)
             break
 
+          case FilingNames.CHANGE_OF_ADDRESS:
+          case FilingNames.CHANGE_OF_DIRECTORS:
+            if (this.isBenBcCccUlc) {
+              // resume correction via Edit UI if current type is BC, CC, ULC, or BEN
+              const correctionUrl = `${this.getEditUrl}${this.getIdentifier}/correction/?correction-id=${item.filingId}`
+              navigate(correctionUrl)
+              break
+            } else {
```

Comment:
This one's messier and does bother me 😁 

How about implementing a local function that you can call, eg.
```
/** Resumes a draft filing. */
protected doResumeFiling (item: TodoItemIF): void {
  ...
  case XXX
    GoToEditUi()
    break
  case YYY
    if (...) {
      GoToEditUi()
      break
    }

  function GoToEditUi (): void {
     ...
  }
}
```

Reply:
Incredible idea Sev. I will be implementing that, thanks a bunch!